spec/services/npq/create_participant_outcome_spec.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe NPQ::CreateParticipantOutcome, :with_default_schedules do
+  let(:cpd_lead_provider) { create(:cpd_lead_provider, :with_lead_provider, :with_npq_lead_provider) }
+  let(:npq_lead_provider) { cpd_lead_provider.npq_lead_provider }
+  let(:npq_application) { create(:npq_application, :eligible_for_funding, :accepted, npq_lead_provider:) }
+  let(:participant_profile) { npq_application.profile }
+  let(:participant_declaration) { create(:npq_participant_declaration, participant_profile:, cpd_lead_provider:) }
+  let(:participant_external_id) { participant_profile.participant_identity.external_identifier }
+  let(:course_identifier) { npq_application.npq_course.identifier }
+  let(:state) { "passed" }
+  let(:completion_date) { "2022-11-30" }
+
+  let(:params) do
+    {
+      cpd_lead_provider:,
+      participant_external_id:,
+      course_identifier:,
+      state:,
+      completion_date:,
+    }
+  end
+
+  subject(:service) { described_class.new(params) }
+
+  describe "#call" do
+    context "when the completion date is missing" do
+      let(:completion_date) {}
+
+      it "is invalid returning a meaningful error message" do
+        is_expected.to be_invalid
+
+        expect(service.errors.messages_for(:completion_date)).to include("The property '#/completion_date' must be present")
+      end
+    end
+
+    context "when the completion date is in the future" do
+      let(:completion_date) { Date.tomorrow.strftime("%F") }
```

Comment:
Isn't all this validation already tested at the model level? Maybe we should remove it from there.

Reply:
Hi @peteglondon Yeah I'm a bit over validating things here.. happy to remove it from here as the model is already validating that.